contractcourt/chain_arbitrator.go

```diff
@@ -763,20 +765,75 @@ func (c *ChainArbitrator) Start() error {
 
 	// Launch all the goroutines for each arbitrator so they can carry out
 	// their duties.
+	// Set a timeout for the group of goroutines. Maybe have different
+	// timeouts for itests and normal operations.
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+
+	// Create an errgroup with the context
+	errGroup, ctx := errgroup.WithContext(ctx)
+
 	for _, arbitrator := range c.activeChannels {
 		startState, ok := startStates[arbitrator.cfg.ChanPoint]
 		if !ok {
 			stopAndLog()
+			// In case we encounter an error we need to cancel the
+			// context to ensure all goroutines are cleaned up.
+			cancel()
 			return fmt.Errorf("arbitrator: %v has no start state",
 				arbitrator.cfg.ChanPoint)
 		}
 
-		if err := arbitrator.Start(startState); err != nil {
-			stopAndLog()
-			return err
-		}
+		errGroup.Go(func() error {
+			// Create buffered error channel for this specific
+			// arbitrator
+			errChan := make(chan error, 1)
+
+			// Start arbitrator in a separate goroutine
+			go func() {
+				errChan <- arbitrator.Start(startState)
+			}()
+
+			// Wait for either the arbitrator to complete or the
+			// context to be cancelled/timing out.
+			select {
+			case err := <-errChan:
+
+				return err
+
+			case <-ctx.Done():
+
+				return ctx.Err()
+			}
+		})
 	}
 
+	// Wait for all arbitrators to start in a separate goroutine. We don't
+	// have to wait here for the chain arbitrator to start, because there
+	// might be situations where other subsystems will block the start up
+	// while fetching resolve information (e.g. custom channels.)
+	//
+	// NOTE: We do not add this collector to the waitGroup because we want
+	// to stop the chain arbitrator if there occurs an error.
+	go func() {
+		defer cancel()
+		select {
+		// As soon as the context cancels we can be sure the
+		// errGroup has finished waiting.
+		case <-ctx.Done():
```

Comment:
I'm not sure we can rely on the context being canceled here. The Godoc for the err group says "the first call to return a non-nil error cancels the group's context". But if there is no failure, there won't be a cancel.

Perhaps instead of using an err group we just create an error channel for the number of arbitrators we start, start them all in a goroutine then wait for them to be completed here, by reading the number of `err` (or `nil`) from the channel as there are arbitrators.